app/models/game.rb

```diff
@@ -46,4 +46,12 @@ def render_piece(x, y)
     piece = Piece.find_by(game_id: id, x_position: x, y_position: y)
     piece.render if piece.present?
   end
+
+  def check?(is_white)
+    king = Piece.find_by(game_id: id, type: KING, is_white: is_white)
```

Comment:
Could also be written as `king = pieces.find_by(type: KING, is_white: is_white)`